rtc_exec_enclave/src/sample_functions.rs

```diff
@@ -0,0 +1,74 @@
+use std::boxed::Box;
+use std::vec::Vec;
+
+use crate::types::*;
+use core::slice;
+pub(crate) trait SampleExecModule {
+    fn call_safe(&self, dataset: &[u8]) -> ExecResult;
+}
+
+impl<T> ExecModule for T
+where
+    T: SampleExecModule,
+{
+    unsafe fn call(&self, dataset_ptr: *const u8, dataset_len: usize) -> ExecResult {
+        let dataset = unsafe { slice::from_raw_parts(dataset_ptr, dataset_len) };
+        self.call_safe(dataset)
+    }
+}
+
+pub(crate) const SHA256_HASH_MODULE_ID: [u8; 32] = [1u8; 32];
+pub(crate) struct Sha256HashModule;
+
+impl SampleExecModule for Sha256HashModule {
+    fn call_safe(&self, dataset: &[u8]) -> ExecResult {
+        match sgx_tcrypto::rsgx_sha256_slice(dataset) {
+            Ok(res) => Ok(res.to_vec().into_boxed_slice()),
+            Err(_) => Err(()),
+        }
+    }
+}
+
+pub(crate) const MEDIAN_MODULE_ID: [u8; 32] = [2u8; 32];
+pub(crate) struct MedianModule;
+
+impl SampleExecModule for MedianModule {
+    fn call_safe(&self, dataset: &[u8]) -> ExecResult {
+        let mut float_dataset: Vec<f64> = dataset
+            .array_chunks()
+            .map(|x| f64::from_ne_bytes(*x))
+            .collect();
+
+        let median = median(&mut float_dataset).ok_or(())?;
+        let median_bytes = median.to_ne_bytes();
+
+        Ok(Box::new(median_bytes))
+    }
+}
+
+fn median(data: &mut [f64]) -> Option<f64> {
+    let len = data.len();
+    // If len is 0 we cannot calculate a median
+    if len == 0 {
+        return None;
+    };
+
```

Comment:
We should probably limit the function to well-defined finite input, and bail out otherwise?

This also means we can drop the `total_cmp` feature, which doesn't make as much numerical sense for the purposes of `median`.

```suggestion

    // No well-defined median if data contains infinities or NaN.
    if !data.iter().all(|n| n.is_finite()) {
        return None;
    }

```